src/Sylius/Bundle/PayumBundle/Validator/GroupsGenerator/GatewayConfigGroupsGenerator.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\PayumBundle\Validator\GroupsGenerator;
+
+use Sylius\Component\Core\Model\PaymentMethodInterface;
+use Symfony\Component\Form\FormInterface;
+use Webmozart\Assert\Assert;
+
+final class GatewayConfigGroupsGenerator
```

Comment:
Maybe we need an interface for this, or at the very least unfinalize it, we could make it `@internal` instead